### PR TITLE
Set up CI based on Travis and OAS Validator, fix #6

### LINKED ISSUE #6: Set up CI
We should set up a CI job to validate the OAS spec, similarly to what's done in other repos in ga4gh-discovery.

### PR DESCRIPTION
**Pull request by mcupak**
This is the CI setup other ga4gh-discovery repositories are using. The script performs validation of the specification file based on Swagger's validator-badge tool.

### CODE REVIEW

**Comment by Relequestual** [comment]
Spectral by Stoplight would provide additional linting with configurable rules. Having said that, I'm not currently sure how to implement that with travis CI, but it's a thought for a later date.
https://stoplight.io/open-source/spectral/

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+sudo: false
+jdk:
+  - openjdk11
+env:
+  - GH_URL=https://raw.githubusercontent.com FILE_TO_VALIDATE=spec/search-api.yaml URL_TO_VALIDATE=$GH_URL/${TRAVIS_PULL_REQUEST_SLUG:-$TRAVIS_REPO_SLUG}/${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH}/$FILE_TO_VALIDATE
+before_install:
+  - git clone --branch=v1.0.1 https://github.com/mcupak/oas-validator.git
+  - ./oas-validator/setup-validator.sh
+script:
+  - ./oas-validator/validate.sh "$URL_TO_VALIDATE"
```
Comment (philsturgeon) [inline .travis.yml:12]: RE chat on twitter, if you wanted spectral you could do:

```suggestion
  - npx @stoplight/spectral lint "$URL_TO_VALIDATE"
```

Then just delete the `before_install` and you're all set.

**Comment by philsturgeon** [comment]
You would eventually probably want your own ruleset but you can use the default ones in Stoplight to start off with. it covers structural validation, semantic validation, and a few other things like raising warnings when it spots <script> tags in the Markdown.

**Comment by mcupak** [comment]
Great find, that tool looks very useful! Thanks!

**Comment by Relequestual** [comment]
It's my long term plan for making a JSON Schema linter tool... =]

**Pull request merged**